tests/test_core/test_new_alignment.py

```diff
@@ -0,0 +1,420 @@
+def test_iter_seq_view(sd_demo: SeqData, idx, seq):
+    got = tuple(sd_demo.iter_seq_view())
+    assert len(got) == 2
+    assert got[idx].seq == sd_demo
+    assert got[idx].seqid == seq
+
+
+@pytest.mark.parametrize(
+    "seq, moltype_name",
+    [("TCAG-NRYWSKMBDHV?", "dna"), ("UCAG-NRYWSKMBDHV?", "rna")],
+)
+def test_seq_index_str(seq, moltype_name, int_arr):
+    alpha = get_moltype(moltype_name).alphabets.degen_gapped
+    got = seq_index(seq, alpha)
+    assert numpy.array_equal(got, int_arr)
+
+
+@pytest.mark.parametrize(
+    "seq, moltype_name",
+    [(b"TCAG-NRYWSKMBDHV?", "dna"), (b"UCAG-NRYWSKMBDHV?", "rna")],
+)
+def test_seq_index_bytes(seq, moltype_name, int_arr):
+    alpha = get_moltype(moltype_name).alphabets.degen_gapped
+    got = seq_index(seq, alpha)
+    assert numpy.array_equal(got, int_arr)
+
+
+@pytest.mark.parametrize("moltype_name", ("dna", "rna"))
+def test_seq_index_arr(moltype_name, int_arr):
+    alpha = get_moltype(moltype_name).alphabets.degen_gapped
+    got = seq_index(int_arr, alpha)
+    assert numpy.array_equal(got, int_arr)
+    assert got.dtype == int_arr.dtype
+
+
+def test_get_seq_array(simple_dict, alpha):
+    # TODO: slicing should be tested here, not get_seq_str
+    # seq1
+    expect = numpy.array([2, 1, 3, 0], dtype="uint8")
+    sd = SeqData(data=simple_dict, alphabet=alpha)
+    got = sd.get_seq_array(seqid="seq1")
+    assert numpy.array_equal(got, expect)
+
+    # seq2
+    expect = numpy.array([3, 0, 0, 0, 3, 1, 2], dtype="uint8")
+    sd = SeqData(data=simple_dict, alphabet=alpha)
+    got = sd.get_seq_array(seqid="seq2")
+    assert numpy.array_equal(got, expect)
+
+
+def test_get_seq_bytes(sd_demo: SeqData):
+    # getting seqid and slicing tested in test_get_seq_str
+    got = sd_demo.get_seq_bytes(seqid="seq1")
+    assert isinstance(got, bytes)
+
+
+# SeqData set_seq_maker tests
+def test_set_set_seq_maker_default(sd_demo):
+    assert sd_demo.set_seq_maker is None
+
+
+@pytest.mark.parametrize(
+    "maker", (get_moltype("dna").make_seq, get_moltype("dna").make_array_seq)
+)
+def test_set_seq_maker_setget(sd_demo, maker):
+    alpha = sd_demo._alphabet
+    sd_demo.set_seq_maker = (maker, alpha)
+    assert sd_demo.set_seq_maker == maker
+
+
+def test_set_seq_maker_wrong_alpha(sd_demo):
+    mt = get_moltype("dna")
+    maker = mt.make_seq
+    alpha = mt.alphabet
+    with pytest.raises(ValueError):
+        sd_demo.set_seq_maker = maker, alpha
+
+
+# SeqData __getitem__ tests
+@pytest.mark.parametrize("seq", ("seq1", "seq2"))
+def test_getitem_str(sd_demo, seq):
+    got = sd_demo[seq]
+    assert got.seq == sd_demo
+    assert got.seqid == seq
+
+
+@pytest.mark.parametrize("idx", (0, 1))
+def test_getitem_int(simple_dict, alpha, idx):
+    sd = SeqData(simple_dict, alphabet=alpha)
+    got = sd[idx]
+    assert got.seq == sd
+    assert got.seqid == list(simple_dict)[idx]
+
+
+# @pytest.mark.parametrize("seqid", ("seq1", "seq2"))
```

Comment:
can mark xfail if failing or delete, don't want to accidentally commented out code to sneak into develop